src/ray/raylet/main.cc

```diff
@@ -121,15 +123,47 @@ absl::flat_hash_map<std::string, std::string> parse_node_labels(
 }
 
 int main(int argc, char *argv[]) {
+  gflags::ParseCommandLineFlags(&argc, &argv, true);
+
+  // TODO(hjiang): For the current implementation, we assume all logging are managed by
+  // spdlog, the caveat is there could be there's writing to stdout/stderr as well. The
+  // final solution is implement self-customized sink for spdlog, and redirect
+  // stderr/stdout to the file descritor. Hold until it's confirmed necessary.
+  //
+  // Backward compatibility notes:
+  // By default, GCS server flushes all logging and stdout/stderr to a single file called
+  // `gcs_server.out`, without log rotations. To keep backward compatibility at best
+  // effort, we use the same filename as output, and disable log rotation by default.
+
+  // For compatibility, by default GCS server dumps logging into a single file with no
+  // rotation.
+  const char *rotation_max_bytes = std::getenv("RAY_ROTATION_MAX_BYTES");
+  // The callchain for java workload is,
+  // - Java runtime creates cluster and runs C++ binaries (i.e. gcs, raylet, etc)
+  // directly;
+  // - For all Java tasks/tasks, they are executed as subprocess in C++ core worker;
+  // - So there's interdepenency for env variables between C++ and Java runtime;
+  // - To keep the logging related env work, use a special env variable
+  // `JAVA_MANAGED_LOGGING` so raylet C++ side knows don't update any logging related
+  // params themselves.
+  const char *java_use_direction = std::getenv("JAVA_MANAGED_LOGGING");
```

Comment:
Don't do this way, pass the default rotation size to StartRayLog.

Reply:
Updated.